### ISSUE TITLE
Zotmoov „Move selected to directory“ does not work for subdirectories

### ISSUE THREAD

**Issue by apfelpfluecker** [wileyyugioh/zotmoov#63]
If I use Zotero with the Firefox add on and add a full-book via SpringerLink to my library, Zotmoov move the PDF to the correct directory but not into a subdirectory.
That doesn’t work neither automatically nor manually. 

![Bildschirmfoto vom 2024-11-01 17-47-49](https://github.com/user-attachments/assets/759596e9-6608-4c05-826b-c342ce4ac0b2)
![Bildschirmfoto vom 2024-11-01 17-48-52](https://github.com/user-attachments/assets/f220065a-f065-4227-b084-7ec65e26c635)

Zotero: 7.0.8 (Linux)
Zotmoov: 1.1.14

**Comment by wileyyugioh** [comment]
Can you share some info or a screenshot about the parent item? My immediate thought is that it doesn't have the author tag filled out.

**Comment by elisaf8** [comment]
Hello, did you happen to solve this issue? It still doesn't work when using even the default %c but possibly it's me doing something wrong.

**Comment by wileyyugioh** [comment]
@elisaf8 can you share some more information? Like a screenshot of the settings page and the item that doesn’t work.

**Comment by elisaf8** [comment]
@wileyyugioh Hello, thank you so much for the prompt response. I am attaching a couple pictures about the situation. I read the documentation but I am not the most tech-savvy so I possibly understood it wrongly. What I would like to have is, similarly to a custom directory where the new added files are saved, a tree-like structure where inside my chosen directory the corresponding subdirectories are created depending on where I put new attachments. 

I don't know if my request is clear and if the plugin is actually able to do that, but I had found online that with the %c command it should work like that. The problem is, if I add a new attachment in a subcollection with the setting "automatically move to subdirectory" turned on, it doesn't move them. For worse, it does not even save a linked copy of the file I attached (image 3). 

I hope my message is clear, and I hope what I am trying to accomplish is possible.

![image](https://github.com/user-attachments/assets/0c7d5e9a-81b5-496d-a259-01366f816333)
![image](https://github.com/user-attachments/assets/d3c3402a-9dd9-4adf-9f14-0c9f4edc41b7)
![image](https://github.com/user-attachments/assets/0a5e8579-3873-428e-aa1c-ad822e7d75f8)

**Comment by wileyyugioh** [comment]
@elisaf8 Does it only not work for the subdirectories? Or for all files? Also, can you tell me the length of your file path? Windows has a limit of 260 characters that has caused problems in the past.

**Comment by wileyyugioh** [comment]
@elisaf8 Can you open up the error console (Tools > Developer > Error Console) and then try and use the menu item to see what errors pop up? It should be in red.

**Comment by wileyyugioh** [comment]
@elisaf8 I am honestly stumped. Can you try other wildcard tags (maybe %a)? Also is this for every collection or just one in particular.

**Comment by elisaf8** [comment]
@wileyyugioh I have tried both with different wildcards and with different collections, doesn't work. It is supposed to create a subfolder named with the desired wildcard, is it right?

**Comment by wileyyugioh** [comment]
@elisaf8 Yes, that's right. Can you try the following command? Select an attachment in Zotero and then enter the code into the error console.

```
Zotero.ZotMoov._getCopyPath(Zotero.getActiveZoteroPane().getSelectedItems()[0], '/', Zotero.ZotMoov.getBasePrefs())
```

You might need to change ```'/'``` with ```'C:/'``` or something like that. What the code does is generate the file path with the wildcards without moving any file.

**Comment by elisaf8** [comment]
@wileyyugioh here you have it, I tried fidgeting around with names but always gives me errors. Maybe the issue is that the file path is not on locale but on OneDrive?

![image](https://github.com/user-attachments/assets/d68a5ad2-0752-4987-8198-51f24044f527)

**Comment by elisaf8** [comment]
@wileyyugioh Tried both! Possibly the issue is the first one you highlighted, because Zootmove includes in the file path also the name of the file itself, which means that the majority of my file paths are longer than 260 characters. It now works with shorter directories... I'll try using a shorter path.

Thank you so much for helping!!

**Comment by apfelpfluecker** [comment]
> Can you share some info or a screenshot about the parent item? My immediate thought is that it doesn't have the author tag filled out.

I’m sorry for answering late, I’ve lost sight of the issue.
Yes, the author tag is not filled out because there are not authors but editors…
![Bildschirmfoto vom 2025-01-10 23-35-27](https://github.com/user-attachments/assets/9b7ec744-3dea-4058-bc14-9ba8f2e62a2f)

I will check if this is always the reason. In past I just had this problem with SpringerLink but it could be the same.

**Comment by wileyyugioh** [comment]
@apfelpfluecker you can use the or option for wildcards so that if the authors tag is blank it will use the editors wildcard instead.

**Comment by apfelpfluecker** [comment]
Ah, ok. Am I right that for my case `{(%a | %d)}` is the correct wildcard?
